From 7cf02ccf402ff92447359b6e67af7e5d35f7664a Mon Sep 17 00:00:00 2001
From: Niall Pemberton <niallp@apache.org>
Date: Sun, 12 Nov 2006 15:21:35 +0000
Subject: [PATCH] BEANUTILS-49 - Remove unnecessary synchronization from the
 static getInstance() and setInstance() methods in BeanUtilsBean and
 LocaleBeanUtilsBean - the ContextClassLoaderLocale that they access already
 has all of its methods synchronized. Thanks to Jesper Richter-Reichhelm for
 reporting this and Kenneth Xu for pointing out the unnecessary
 synchronization.

git-svn-id: https://svn.apache.org/repos/asf/jakarta/commons/proper/beanutils/trunk@473972 13f79535-47bb-0310-9956-ffa450edef68
---
 src/java/org/apache/commons/beanutils/BeanUtilsBean.java      | 4 ++--
 .../apache/commons/beanutils/locale/LocaleBeanUtilsBean.java  | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/java/org/apache/commons/beanutils/BeanUtilsBean.java b/src/java/org/apache/commons/beanutils/BeanUtilsBean.java
index d66bf6932..116322ace 100644
--- a/src/java/org/apache/commons/beanutils/BeanUtilsBean.java
+++ b/src/java/org/apache/commons/beanutils/BeanUtilsBean.java
@@ -75,7 +75,7 @@ protected Object initialValue() {
      *
      * @return The (pseudo-singleton) BeanUtils bean instance
      */
-    public synchronized static BeanUtilsBean getInstance() {
+    public static BeanUtilsBean getInstance() {
         return (BeanUtilsBean) beansByClassLoader.get();
     }
 
@@ -86,7 +86,7 @@ public synchronized static BeanUtilsBean getInstance() {
      * 
      * @param newInstance The (pseudo-singleton) BeanUtils bean instance
      */
-    public synchronized static void setInstance(BeanUtilsBean newInstance) {
+    public static void setInstance(BeanUtilsBean newInstance) {
         beansByClassLoader.set(newInstance);
     }
 
diff --git a/src/java/org/apache/commons/beanutils/locale/LocaleBeanUtilsBean.java b/src/java/org/apache/commons/beanutils/locale/LocaleBeanUtilsBean.java
index fac243f5a..4c1cbf005 100644
--- a/src/java/org/apache/commons/beanutils/locale/LocaleBeanUtilsBean.java
+++ b/src/java/org/apache/commons/beanutils/locale/LocaleBeanUtilsBean.java
@@ -57,7 +57,7 @@ protected Object initialValue() {
                     };
      
      /** Gets singleton instance */
-     public synchronized static LocaleBeanUtilsBean getLocaleBeanUtilsInstance() {
+     public static LocaleBeanUtilsBean getLocaleBeanUtilsInstance() {
         return (LocaleBeanUtilsBean)localeBeansByClassLoader.get();
      }
  
@@ -66,7 +66,7 @@ public synchronized static LocaleBeanUtilsBean getLocaleBeanUtilsInstance() {
      * This is a pseudo-singleton - an single instance is provided per (thread) context classloader.
      * This mechanism provides isolation for web apps deployed in the same container. 
      */
-    public synchronized static void setInstance(LocaleBeanUtilsBean newInstance) {
+    public static void setInstance(LocaleBeanUtilsBean newInstance) {
         localeBeansByClassLoader.set(newInstance);
     }
 
